### PR TITLE
Add support for custom configuration

### PR DESCRIPTION
**Pull request by MartinBasti**
Signed-off-by: Martin Bašti <mbasti@redhat.com>

### CODE REVIEW

File: omps/settings.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                    config_file, e
+                )
+            )
+        else:
+            config_section_obj = getattr(config_module, config_section, None)
+            if config_section_obj is None:
+                raise RuntimeError('No section "{}" in "{}" found!'.format(
+                    config_section, config_file
+                ))
+    elif any('py.test' in arg or 'pytest.py' in arg for arg in sys.argv):
```
Comment (alexandrevicenzi) [inline omps/settings.py]: pytest is also available and should be considered.

Also, what about some ENV variable to detect if we are under CI?

File: omps/settings.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+class TestConfig(DefaultConfig):
+    TESTING = True
+
+
+def init_config(app):
+    """Configure OMPS"""
+    config_section = 'ProdConfig'
+    config_section_obj = ProdConfig
+
+    if constants.ENV_CONF_SECTION in os.environ:
```
Comment (alexandrevicenzi) [inline omps/settings.py]: config_section = os.getenv(constants.ENV_CONF_SECTION, 'ProdConfig')

File: omps/settings.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                    config_section, config_file
+                ))
+    elif any('py.test' in arg or 'pytest.py' in arg for arg in sys.argv):
+        config_section_obj = TestConfig
+    elif constants.ENV_DEVELOPER_ENV in os.environ:
+        config_section_obj = DevConfig
+
+    conf = Config(config_section_obj)
+    app.config.from_object(config_section_obj)
+    return conf
```
Comment (alexandrevicenzi) [inline omps/settings.py]: Why return a value if it's never used?

File: omps/settings.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        with runtime values.
+        """
+
+        # set defaults
+        for name, values in self._defaults.items():
+            self.set_item(name, values['default'])
+
+        # override defaults
+        for key in dir(conf_section_obj):
+            # skip keys starting with underscore
```
Comment (alexandrevicenzi) [inline omps/settings.py]: These types of comments can be removed. If you read the code it's easy to understand.

File: setup.py [hunk 5]
```diff
@@ -36,6 +36,7 @@
     python_requires='>=3.6, <4',
     install_requires=[
         'Flask==1.0.*',
+        'PyYAML',
```
Comment (alexandrevicenzi) [inline setup.py]: Where is this being used?

Comment (MartinBasti) [reply setup.py]: sorry leftover, will remove it

File: omps/settings.py (hunk 4 above)
Comment (MartinBasti) [reply omps/settings.py]: I just copied this code from freshmaker

Comment (alexandrevicenzi) [reply omps/settings.py]: ok

File: omps/settings.py (hunk 3 above)
Comment (MartinBasti) [reply omps/settings.py]: Hopefully this will be used in later PRs

File: omps/settings.py (hunk 1 above)
Comment (MartinBasti) [reply omps/settings.py]: changed to `pytest`, this should cover also pytest.py

Comment (MartinBasti) [reply omps/settings.py]: about the CI: not now, so far CI runs only pytest

**Comment by MartinBasti** [comment]
PR updated

File: omps/settings.py (hunk 3 above)
Comment (alexandrevicenzi) [reply omps/settings.py]: :thinking:

**Pull request merged**